File: packages/expo-image/android/src/main/java/expo/modules/image/events/ImageLoadStartEvent.java
package expo.modules.image.events;

import com.facebook.react.bridge.Arguments;
import com.facebook.react.uimanager.events.Event;
import com.facebook.react.uimanager.events.RCTEventEmitter;

public class ImageLoadStartEvent extends Event<ImageLoadStartEvent> {
  public static final String EVENT_NAME = "onLoadStart";

  public ImageLoadStartEvent(int viewId) {
    super(viewId);
  }

  @Override
  public String getEventName() {
    return EVENT_NAME;
  }

  @Override
  public void dispatch(RCTEventEmitter rctEventEmitter) {
    rctEventEmitter.receiveEvent(getViewTag(), getEventName(), Arguments.createMap());
  }
}

File: packages/expo-image/android/src/main/java/expo/modules/image/events/ImageLoadStartEvent.kt
package expo.modules.image.events

import com.facebook.react.bridge.Arguments
import com.facebook.react.uimanager.events.Event
import com.facebook.react.uimanager.events.RCTEventEmitter

class ImageLoadStartEvent(viewId: Int) : Event<ImageLoadStartEvent>(viewId) {
  override fun getEventName() = EVENT_NAME

  override fun dispatch(rctEventEmitter: RCTEventEmitter) {
    rctEventEmitter.receiveEvent(viewTag, eventName, Arguments.createMap())
  }

  companion object {
    const val EVENT_NAME = "onLoadStart"
  }
}
